streams/src/test/java/org/apache/kafka/test/MockInternalNewProcessorContext.java

```diff
@@ -41,12 +41,16 @@
 import org.apache.kafka.streams.state.internals.ThreadCache.DirtyEntryFlushListener;
 
 import java.io.File;
+import java.util.LinkedHashMap;
+import java.util.Map;
 import java.util.Objects;
 import java.util.Properties;
 
 public class MockInternalNewProcessorContext<KOut, VOut> extends MockProcessorContext<KOut, VOut> implements InternalProcessorContext<KOut, VOut> {
 
-    private ProcessorNode currentNode;
+    private ProcessorNode<?, ?, ?, ?> currentNode;
+    private RecordCollector recordCollector;
```

Comment:
Add missing `recordCollector` "feature -- the old `MockInternalProcessorContext` that we remove with this PR had this, and the test we rewrite use this feature